Lib/test/test_clinic.py

```diff
@@ -1692,5 +1692,167 @@ def test_permute_optional_groups(self):
                 self.assertEqual(actual, expected)
 
 
+class FormatHelperTests(unittest.TestCase):
+
+    def test_strip_leading_and_trailing_blank_lines(self):
+        dataset = (
+            # Input lines, expected output.
+            ("a\nb",            "a\nb"),
+            ("a\nb\n",          "a\nb"),
+            ("a\nb ",           "a\nb"),
+            ("\na\nb\n\n",      "a\nb"),
+            ("\n\na\nb\n\n",    "a\nb"),
+            ("\n\na\n\nb\n\n",  "a\n\nb"),
+            # Note, leading whitespace is preserved:
+            (" a\nb",               " a\nb"),
+            (" a\nb ",              " a\nb"),
+            (" \n \n a\nb \n \n ",  " a\nb"),
+        )
+        for lines, expected in dataset:
+            with self.subTest(lines=lines, expected=expected):
+                out = clinic.strip_leading_and_trailing_blank_lines(lines)
+                self.assertEqual(out, expected)
+
+    def test_normalize_snippet(self):
+        snippet = """
+            one
+            two
+            three
+        """
+        template = (
+            "{indent}one\n"
+            "{indent}two\n"
+            "{indent}three"
+        )
+        for indent in 0, 4, 8:
+            expected = template.format(indent=" " * indent)
+            with self.subTest(snippet=snippet, indent=indent, expected=expected):
+                out = clinic.normalize_snippet(snippet, indent=indent)
+                self.assertEqual(out, expected)
```

Comment:
This took me a little bit of time to grok; perhaps it's a little too clever to compute the expected result in the test? Maybe something a little more verbose and a little more dumb might be better here, since this _is_ a test after all :)

```suggestion
        zero_indent = (
            "one\n"
            "two\n"
            "three"
        )
        four_indent = (
            "    one\n"
            "    two\n"
            "    three"
        )
        eight_indent = (
            "        one\n"
            "        two\n"
            "        three"
        )
        expected_outputs = {0: zero_indent, 4: four_indent, 8: eight_indent}

        for indent_length, expected_output in expected_outputs.items():
            with self.subTest(indent_length=indent_length):
                actual_output = clinic.normalize(snippet, indent=indent_length)
                self.assertEqual(actual_output, expected_output)
```